vm/src/obj/objbyteinner.rs

```diff
@@ -94,7 +98,148 @@ impl PyByteInner {
             }
         }
     }
+}
+
+#[derive(FromArgs)]
+pub struct ByteInnerFindOptions {
+    #[pyarg(positional_only, optional = false)]
+    sub: PyObjectRef,
+    #[pyarg(positional_only, optional = true)]
+    start: OptionalArg<PyObjectRef>,
```

Comment:
This can be `PyIntRef`.

Reply:
It could be  Int or None so I have to accept PyObjectRef

Reply:
Can you do `OptinalArg<PyIntRef>`?

Reply:
Or maybe `OptionalArg<Option<PyIntRef>>`? I'm not sure if `Option` is `FromArgs` yet, but it should be the specified type or Python `None`.

Reply:
using  `OptionalArg<Option<PyIntRef>>`, you mean this implementation in impl  block (line 126)?
```
let start = match self.start {
            OptionalArg::Present(Some(int)) => Some(int.as_bigint().clone()),
            _ => None,
        };
```

Reply:
I think you can use `into_option` of `OptionalArg` to make this code a little nicer but this looks good. This will move all the type checks to the `FromArgs`.

Reply:
I let it  for now. 
```rust
let start = match self.start {
            OptionalArg::Present(Some(int)) => Some(int.as_bigint().clone()),
            _ => None,
        };
```
instead of 
```rust
let start = match self.start.into_option() {
            Some(Some(int)) => Some(int.as_bigint().clone()),
            _ => None,
        };
```

using into_option() isn't much shorter and I find it less explicit. what do you think ?

Reply:
They both look great to me.

Reply:
The rust clippy tool would probably suggest to use something like this:
```rust
let start = if let Some(Some(int)) = self.start.into_option() {
   Some(int.as_bigint().clone())
} else {
   None
};
```

Reply:
I use clippy and non suggestion about it

Reply:
It's especially handy when you have a match with only two options (basically an if-else case).